patches/minecraft/net/minecraft/server/MinecraftServer.java.patch

```diff
@@ -172,26 +172,45 @@
  
        try {
           OptionSet optionset = optionparser.parse(p_main_0_);
-@@ -910,6 +935,10 @@
+@@ -888,15 +913,16 @@
+ 
+          Path path = Paths.get("server.properties");
+          ServerPropertiesProvider serverpropertiesprovider = new ServerPropertiesProvider(path);
+-         serverpropertiesprovider.func_219035_b();
+          Path path1 = Paths.get("eula.txt");
+          ServerEula servereula = new ServerEula(path1);
+          if (optionset.has(optionspec1)) {
++            serverpropertiesprovider.func_219035_b();
+             field_147145_h.info("Initialized '" + path.toAbsolutePath().toString() + "' and '" + path1.toAbsolutePath().toString() + "'");
+             return;
+          }
+ 
+          if (!servereula.func_154346_a()) {
++            serverpropertiesprovider.func_219035_b();
```

Comment:
Am I correct in thinking that this wipes any changes to the properties if you have forgotten to accept the EULA (which I always forget to do)?
Sorry, it's a bit hard to read patch files on my phone, so I could be wrong about it, but I would rather it didn't affect the file if the EULA hasn't been accepted. Otherwise there's a risk that you change some settings, run the server, accept the EULA then re-run expecting your changed settings to apply and they won't.

Reply:
It'd behave the same as the unmodified code does now; reset the `level-type` property to default if it was set to anything non-vanilla.

Other properties (providing they're valid) remain the same

Reply:
Could we not leave this out?
It would decrease the patch size, leave user changes intact, and any invalid settings would be corrected next run.

Reply:
I'd assume that for most people's first run of a server there wouldn't be a server.properties file and they'd expect it to be generated. I realise that's technically what the init flag is for but this has been the behaviour since I can remember.

A better solution would probably be to check if the file exists before deciding whether to write it or not, but would increase the patch size by a couple of lines rather than decrease it - not sure how much of a concern that would be.

Reply:
Yeah, wasn't really thinking about that, it probably should be generated even if the EULA hasn't been accepted. For some reason I was thinking that the file was already there, when in most cases it wouldn't be.

Reply:
I think this should definitely not be here anymore, because if it didn't exist it would have been created by the line above, but if it didn't it shouldn't be overwritten just because the EULA isn't accepted.

Reply:
This is outdated, see 2513a10.